app/gui/src/controller/project.rs

```diff
@@ -232,6 +234,35 @@ impl Project {
 }
 
 
+// === Project Snapshotting ===
+
+impl Project {
+    /// Saves a snapshot of the current state of the project to the VCS.
+    #[profile(Detail)]
+    pub fn save_project_snapshot(&self) -> impl Future<Output = FallibleResult> {
+        let project_root_id = self.model.project_content_root_id();
+        let path_segments: [&str; 0] = [];
+        let root_path = Path::new(project_root_id, &path_segments);
+        let language_server = self.model.json_rpc();
+        async move {
+            let response = language_server.save_vcs(&root_path, &None).await;
+            if let Err(RpcError::RemoteError(json_rpc::messages::Error {
+                code: error_code, ..
+            })) = response
+            {
+                if error_code == code::FILE_NOT_FOUND {
```

Comment:
Oh, this is FILE_NOT_FOUND? I need to change that so that it returns `VCSNotFound`.. sorry!